### PR TITLE
Add lesson list search functionality plus a bunch of rubocop auto changes

### PR DESCRIPTION
**Pull request by jmkoni**

### CODE REVIEW

File: app/models/lesson.rb [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+  ARTICLE = 'Article-based'
+  LESSON_TYPES = [ACTIVITY, ARTICLE].freeze
+
+  has_many :lesson_terms
+  has_many :terms, through: :lesson_terms
+
   validates_presence_of :body, :lesson_type, :questions, :title
   validates_inclusion_of :lesson_type, in: LESSON_TYPES
+
+  pg_search_scope :search_by_title, against: :title
```
Comment (almightyboz) [inline app/models/lesson.rb:16]: I didn't know about pg_search, this is cool!

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
+  def edit; end
 
   # POST /lessons
   # POST /lessons.json
   def create
     @lesson = Lesson.new(lesson_params)
     respond_to do |format|
       if @lesson.save
-        puts "saved"
+        puts 'saved'
```
Comment (osondoar) [inline app/controllers/lessons_controller.rb]: We can remove the `puts` if it's no longer needed for testing.

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -1,37 +1,49 @@
...
         format.html { redirect_to @lesson, notice: 'Lesson was successfully created.' }
         format.json { render :show, status: :created, location: @lesson }
       else
-        puts "NOPE"
+        puts 'NOPE'
```
Comment (osondoar) [inline app/controllers/lessons_controller.rb]: same here.

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,20 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+10.times do
+  Term.create(name: Faker::Hipster.word)
```
Comment (almightyboz) [inline db/seeds.rb:12]: Personal opinion: I don't think that Faker::Hipster is funny. But I've been accused of being a hipster myself, soooo  https://media.giphy.com/media/xTiTnsHTXnBheAIM5G/giphy.gif

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,20 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
```
Comment (almightyboz) [inline db/seeds.rb:10]: Would it be advantageous for the first lines to destroy all existing term and lesson instances? I sometimes double-seed by accident, and it prevents errors like this: http://stackoverflow.com/questions/12103333/rails-reset-seeds-rb .

**Comment by osondoar** [comment]
👍

File: db/seeds.rb (hunk 4 above)
Comment (jmkoni) [reply db/seeds.rb:12]: it's less funny and more it was an easy one for a word. THERE ARE TOO MANY FAKERS NOW UGHHHHHH

**Comment by almightyboz** [comment]
👍 :shipit:

**Pull request merged**